api/proto/http/v1/http.proto

```diff
@@ -0,0 +1,137 @@
+syntax="proto3";
+package http.v1;
+option go_package="github.com/iansmith/parigot/g/http/v1;http";
+import "protosupport/v1/protosupport.proto";
+
+// This is the http service for parigot, which provides the most commonly used
+// HTTP methods: GET, POST, PUT, DELETE, HEAD, OPTIONS, PATCH, CONNECT, TRACE
+
+service Http {
+	// Get should only retrieve data and should have no other effect.
+	rpc Get(GetRequest) returns (GetResponse);
+	// Post is to submit data to be processed to a specified resource. It's
+	// often used when submitting form data or uploading a file.
+	rpc Post(PostRequest) returns (PostResponse);
+	// Put is to update the current resource with the uploaded content.
+	rpc Put(PutRequest) returns (PutResponse);
+	// Delete is to remove the specified resource.
+	rpc Delete(DeleteRequest) returns (DeleteResponse);
+	// Head is similar to Get, but it only requests the headers and not the
+	// actual data. This is useful to check if a resource exists or to retrieve
+	// meta-information.
+	rpc Head(HeadRequest) returns (HeadResponse);
+	// Options describes the communication options for the target resource.
+	// Often used to check which methods are supported by a URL or to use CORS
+	// (Cross-Origin Resource Sharing).
+	rpc Options(OptionsRequest) returns (OptionsResponse);
+	// Patch applies partial modifications to a resource. This method allows
+	// updating a resource by sending only the changes instead of sending the
+	// complete new resource representation.
+	rpc Patch(PatchRequest) returns (PatchResponse);
+	// Connect establishes a network connection to a resource (usually for
+	// network tunneling with SSL).
+	rpc Connect(ConnectRequest) returns (ConnectResponse);
+	// Trace performs a loopback test with the path to the target resource, useful for debugging.
+	rpc Trace(TraceRequest) returns (TraceResponse);
+}
+
+message StringList {
+	repeated string stringlist = 1;
+}
+
+// Common message for fields that are used in method requests
+message HttpRequest {
+	string url = 1;
+	map<string, StringList> header = 2;
```

Comment:
In principle every header and trailer can be mentioned more than once like this:

```
X-my-header: foo
X-my-header: bar
```

I think it is good to be working like the spec (and you are) but if you simplified that to be `map<string,string> I don't think it would be the end of the world.  It's up to you.

Reply:
reasonable, simplified it